src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -171,7 +171,14 @@ private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
                 throw new ArgumentException(CodeAnalysisResources.ArgumentElementCannotBeNull, nameof(analyzers));
             }
 
-            return Create(compilation, analyzers, options, analyzerManager, addExceptionDiagnostic, out newCompilation, continueOnAnalyzerException: null, cancellationToken: cancellationToken);
+            Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException = (ex, analyzer, diagnostic) =>
+            {
+                if (addExceptionDiagnostic != null)
+                {
+                    addExceptionDiagnostic(diagnostic);
+                }
+            };
+            return Create(compilation, analyzers, options, analyzerManager, onAnalyzerException, out newCompilation, cancellationToken: cancellationToken);
```

Comment:
@jaredpar this is the single point where we initialize the onAnalyzerException for compiler invocations into analyzer layer. Command line compiler just passes in _exceptionDiagnosticBag.Add_ delegate for _addExceptionDiagnostic_ parameter, and the compiler analyzer driver wraps it with the delegate needed by the shared AnalyzerExecutor.